src/Resources/ResourceManager/Implementation/CmdletBase/ResourceWithParameterCmdletBase.cs

```diff
@@ -116,69 +116,78 @@ protected ResourceWithParameterCmdletBase()
         [ValidateNotNullOrEmpty]
         public string TemplateUri { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Skips the PowerShell dynamic parameter processing that checks if the provided template parameter contains all necessary parameters used by the template. " +
+                                                    "This check would prompt the user to provide a value for the missing parameters, but providing the -SkipTemplateParameterPrompt will ignore this prompt and " +
+                                                    "error out immediately if a parameter was found not to be bound in the template. For non-interactive scripts, -SkipTemplateParameterPrompt can be provided " +
+                                                    "to avoid generic error messages from being thrown.")]
```

Comment:
To provide a better error message in the case where not all required parameters are satisfied